src/main/java/io/github/opencubicchunks/cubicchunks/core/asm/mixin/core/common/MixinWorldServer_Bukkit.java

```diff
@@ -0,0 +1,434 @@
+import io.github.opencubicchunks.cubicchunks.core.server.ChunkGc;
+import io.github.opencubicchunks.cubicchunks.core.server.CubeProviderServer;
+import io.github.opencubicchunks.cubicchunks.core.server.PlayerCubeMap;
+import io.github.opencubicchunks.cubicchunks.core.server.SpawnCubes;
+import io.github.opencubicchunks.cubicchunks.core.server.VanillaNetworkHandler;
+import io.github.opencubicchunks.cubicchunks.core.util.world.CubeSplitTickList;
+import io.github.opencubicchunks.cubicchunks.core.util.world.CubeSplitTickSet;
+import io.github.opencubicchunks.cubicchunks.core.world.CubeWorldEntitySpawner;
+import io.github.opencubicchunks.cubicchunks.core.world.IWorldEntitySpawner;
+import io.github.opencubicchunks.cubicchunks.core.world.chunkloader.CubicChunkManager;
+import io.github.opencubicchunks.cubicchunks.core.world.cube.Cube;
+import io.github.opencubicchunks.cubicchunks.core.world.provider.ICubicWorldProvider;
+import mcp.MethodsReturnNonnullByDefault;
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.EntityTracker;
+import net.minecraft.entity.effect.EntityLightningBolt;
+import net.minecraft.entity.passive.EntitySkeletonHorse;
+import net.minecraft.init.Blocks;
+import net.minecraft.server.management.PlayerChunkMap;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.DifficultyInstance;
+import net.minecraft.world.NextTickListEntry;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldEntitySpawner;
+import net.minecraft.world.WorldServer;
+import net.minecraft.world.chunk.Chunk;
+import net.minecraft.world.chunk.storage.ExtendedBlockStorage;
+import net.minecraftforge.common.ForgeChunkManager;
+import org.bukkit.event.entity.CreatureSpawnEvent;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Implements;
+import org.spongepowered.asm.mixin.Interface;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * Implementation of {@link ICubicWorldServer} interface.
+ */
+@MethodsReturnNonnullByDefault
+@ParametersAreNonnullByDefault
+@Mixin(WorldServer.class)
+@Implements(@Interface(iface = ICubicWorldServer.class, prefix = "world$"))
+public abstract class MixinWorldServer_Bukkit extends MixinWorld implements ICubicWorldInternal.Server {
```

Comment:
I really don't like this being a completely separate class just for bukkit, as it stands I have no idea what the differences vs normal one are. Is the `addEntity` method the only change? I think it would be better to just call this one through CompatHandler with a MethodHandle